docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -846,6 +846,41 @@ quarkus.http.auth.permission.http-upgrade.paths=/end
 quarkus.http.auth.permission.http-upgrade.policy=authenticated
 ----
 
+Security annotations used during authentication must be placed on an endpoint class as well, for the `SecurityIdentity` is created before the websocket connection is opened.
+
+.Select Bearer token authentication mechanism and OIDC tenant with annotation
+[source, java]
+----
+package io.quarkus.websockets.next.test.security;
+
+import io.quarkus.oidc.BearerTokenAuthentication;
+import io.quarkus.oidc.Tenant;
+import io.quarkus.websockets.next.OnTextMessage;
+import io.quarkus.websockets.next.WebSocket;
+
+@BearerTokenAuthentication <1>
+@Tenant("hr") <2>
```

Comment:
I wonder if it is a bit too OIDC specific for the WebSockets Next guide ? The PR code updates related to it may be fine but I'm not sure we should go into OIDC multi-tenancy is this doc

Reply:
We must make it clear that these annotations should only be used on HTTP upgrade and not on methods like `@OnTextMessage void onTextMessage()` (because it's too late). There is build-time validation that prevents users to use it on methods and also, if this validation somehow failed, there is runtime validation that prevents this as well (so literally impossible for it to happen). but still, I think it is WS Next specific, this is only extension where we use class-level security so far. I don't want to confuse users?

Reply:
to be clear, I am open to any changes, I just think it is very important information, I want users to understand HOW and WHY (both is there - there is text that explain when auth happens)

Reply:
Sure, it is just that is a very OIDC specific content, `BearerTokenAuthentication` can be supported by smallrye-jwt, may be by custom mechanisms, but `@Tenant` is a multi-tenant OIDC specific detail.

Perhaps, you should add a note in the OIDC tenancy doc instead, in the section which describes `@Tenant`, and link to the section you are adding here ?

Reply:
> Sure, it is just that is a very OIDC specific content, `BearerTokenAuthentication` can be supported by smallrye-jwt, may be by custom mechanisms, but `@Tenant` is a multi-tenant OIDC specific detail.

true

> Perhaps, you should add a note in the OIDC tenancy doc instead, in the section which describes `@Tenant`, and link to the section you are adding here ?

I'll do that.

Reply:
Done, I'll resolve this as we can address current state elsewhere. Thanks for the suggestion.